wire-ios/Wire-iOS/Sources/UserInterface/SelfProfile/ProfileHeaderViewController.swift

```diff
@@ -353,6 +332,30 @@ final class ProfileHeaderViewController: UIViewController {
         }
     }
 
+    private func updateE2EICertifiedStatus() {
+        guard let contextProvider = userSession as? ContextProvider else { return }
+        Task {
+            do {
+                let context = contextProvider.syncContext
+                let (selfUser, selfConversation) = await context.perform {
+                    let selfUser = ZMUser.selfUser(in: context)
+                    let mlsSelfConversation = ZMConversation.fetchSelfMLSConversation(in: context)
+                    return (selfUser, mlsSelfConversation)
+                }
+                guard let selfConversation else { return }
+                let isE2EICertified = try await isUserE2EICertifiedUseCase.invoke(
+                    conversation: selfConversation,
+                    user: selfUser
+                )
+                await MainActor.run {
+                    userStatus.isCertified = isE2EICertified
+                }
+            } catch {
+                WireLogger.e2ei.error("failed to get E2EI certification status: \(error)")
+            }
+        }
+    }
```

Comment:
isn't it the same logic as ub ConversationListViewController.ViewModel?

Reply:
Now, `ConversationListViewController.ViewModel` contains only logic for the self user (different use case type).